File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        l = 0
        r = len(heights)-1
        wat = []
        while(l < r):
            water = min(heights[r],heights[l]) * (r-l)
            if(min(heights[l],heights[r]) == heights[l]):
                l = l + 1
            else:
                r = r-1
            wat.append(water)
        return max(wat)
        

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.cpp
class Solution {
public:
    int maxProfit(vector<int>& p) {
        
        int min_price = p[0];
        int max_profit = 0;

        for(int i = 0;i<p.size();i++){
            int c = p[i];
            
            int profit = c - min_price;
            min_price = min(min_price,c);
            max_profit = max(max_profit,profit);
            
    }
    return max_profit;
        
    }
};
